src/components/ui/firstvisittooltip.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import * as TooltipPrimitive from "@radix-ui/react-tooltip";
+import { cn } from "~/utils/helper";
+
+const TooltipProvider = TooltipPrimitive.Provider;
+const TooltipTrigger = TooltipPrimitive.Trigger;
+const TooltipContent = TooltipPrimitive.Content;
+
+const FirstVisitTooltip = ({
+  children,
+  text,
+  showOnFirstVisit = false,
+}: {
+  children: React.ReactNode;
+  text: string;
+  showOnFirstVisit?: boolean;
+}) => {
+  const [shouldRenderTooltip, setShouldRenderTooltip] = React.useState(false);
+
+  React.useEffect(() => {
+    const tooltipShown = localStorage.getItem("tooltipShown") === "true";
+    console.log(
+      `Initial tooltipShown value from localStorage: ${tooltipShown}`
+    );
+
+    if (showOnFirstVisit && !tooltipShown) {
+      setShouldRenderTooltip(true);
+      console.log("Tooltip set to show on first visit");
```

Comment:
`console.log` are good for debugging :slightly_smiling_face: but we shouldn't keep them in production, especially on client side as they are all visible in the browser's console.

I think all the ones in this file can be removed.